### DISCUSSION TITLE
Yarn v2 link specific packages as it works in Yarn v1

### DISCUSSION THREAD

**Discussion by RuBAN-GT** [yarnpkg/berry#2947] [Q&A] [answered]
Hi everyone,

I have a simple scenario to create a symlink from node_modules to specific npm package directory as it works with `npm link package_name` or `yarn link package_name` (yarn v1) commands.
But as I understood properly, Yarn v2 has changes of this command behaviour (https://yarnpkg.com/cli/link) and creates `resolutions` at package.json.

Does somebody advice how to link packages as it was early?

Thank you!

**Comment by larixer** [comment, marked as answer]
There is no CLI command to create symlink in Yarn 2, but this behaviour is still supported. You can use `link:` protocol to achieve this behaviour:
https://yarnpkg.com/features/protocols
`package.json`
```json
{
  "dependencies": {
    "my-package": "link:../relative/path/to/my-package"
  }
}
```

**Reply by RuBAN-GT** [reply]
Hi @larixer. Thank you for your answer. My point is not modify original sources and allow developers to work with it locally without any file changes.

**Reply by RuBAN-GT** [reply]
@larixer, is it possible to call original Yarn CLI command instead of Yarn v2 at project directory (I mean yarn link)?

**Reply by larixer** [reply]
Yes, `YARN_IGNORE_PATH=1 yarn link ...`

**Reply by RuBAN-GT** [reply]
Thank you so much @larixer for your help! What the reason of removing default npm functionality (link)?

**Reply by larixer** [reply]
@RuBAN-GT The short answer is that `yarn link` is now working better than it used to work in `yarn 1` and `npm`, because it installs dependencies from the target package you are linking to, previously it just created symlink and you had to run `yarn install` inside every target folder you link to, now you need to run install only inside your main project. This works the best with the default pnp install strategy. It works with node_modules install strategy too, although with some difficulties, but once you find your way around it, you can see the value in new functionality, it needs some learning.

**Reply by muryoh** [reply]
Interesting discussion :) I'm sort of jumping in unannounced but it's very related
The `unlink` command is still documented but doesn't seem available - is there an way to undo the effect of a link command, particularly when working with the node_modules linker?

**Reply by larixer** [reply]
The `unlink` command will be available on the next publishing round, it has been implemented and merged, but not published yet:
https://github.com/yarnpkg/berry/pull/2588

**Reply by muryoh** [reply]
Very nice, thanks for the update

**Reply by r1y4h** [reply]
> Yes, `YARN_IGNORE_PATH=1 yarn link ...`

Noobie question. What is `YARN_IGNORE_PATH=1` I'm also on the same exact scenario as OP. We only need to use yarn link locally.

Is it run like this inside your pacakge.json script, example:

"link": "YARN_IGNORE_PATH=1 yarn link --private pacakge-1 "

**Reply by merceyz** [reply]
It will ignore the Yarn version specified by your project and invoke the global version (most likely v1) , but you should avoid doing that as much as possible. Use the `yarn link` and `yarn unlink` commands from v2/v3 instead

**Reply by r1y4h** [reply]
Thank you. yarn unlink still seems not available in v2.4.1. And just like what OP says, v1 link was the better choice for what we need as it doesn't alter any code files.

**Reply by merceyz** [reply]
> yarn unlink still seems not available in v2.4.1

It's in the latest v3 RC
```
yarn set version berry && yarn set version canary
```

> And just like what OP says, v1 link was the better choice for what we need as it doesn't alter any code files.

The v1 link wont install dependencies and doesn't work with PnP but if that's fine for you then feel free to use it

**Reply by r1y4h** [reply]
Yeah that's my original problem. I come here looking for why `yarn link` has changed and how it's intended to use now. Seems we have to adopt to the new yarn link now.
